### PR TITLE
[ONPREM-329] Machine Runner macOS docs

### PR DESCRIPTION
**Pull request by circlery**
# Description
Adds machine runner 3.0 preview documentation for macOS systems.

# Reasons
[ONPREM-329]

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

[ONPREM-329]: https://circleci.atlassian.net/browse/ONPREM-329?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: jekyll/_cci2/install-machine-runner-3-on-osx.adoc [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+Next, open the runner config.yaml file with the text editor of your choice, and modify the `[[RUNNER_NAME]]` and `[[RESOURCE_CLASS_TOKEN]]` template values.
+
+```shell
+nano $HOME/Library/Preferences/com.circleci.runner/config.yaml
+```
+
+```
+runner:
+  name: "my-osx-runner"
+  working_directory: "/Users/yourusername/Library/com.circleci.runner/workdir"
```
Comment (jrahme-cci) [inline jekyll/_cci2/install-machine-runner-3-on-osx.adoc:64]: I think we can safely swap `yourusername` for $USER to make it more pastable

**Review by jrahme-cci — APPROVED** [review]

File: jekyll/_cci2/install-machine-runner-3-on-osx.adoc (hunk 1 above)
Comment (circlery) [reply jekyll/_cci2/install-machine-runner-3-on-osx.adoc:64]: good catch.

Comment (circlery) [reply jekyll/_cci2/install-machine-runner-3-on-osx.adoc:64]:
```suggestion
  working_directory: "/Users/$USER/Library/com.circleci.runner/workdir"
```

File: jekyll/_cci2/install-machine-runner-3-on-osx.adoc [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+----
+# remove this block from your existing config
+logging:
+  file: /Library/Logs/com.circleci.runner.log
+----
+
+[#review-accept-apple-signature-notarization]
+== 3. Review and accept the Apple signature notarization
+
+The binary must be approved to run on your macOS system because the self-hosted runner is not compiled from source during installation. This can be done via the UI by accepting the macOS pop-up asking if you wish to run the binary from the internet, or programmatically.
```
Comment (rosieyohannan) [inline jekyll/_cci2/install-machine-runner-3-on-osx.adoc:97]: "via the UI" - what UI are we talking about here? @jrahme-cci @circlery ?

**Review by rosieyohannan — APPROVED** [review]

File: jekyll/_cci2/install-machine-runner-3-on-osx.adoc (hunk 2 above)
Comment (jrahme-cci) [reply jekyll/_cci2/install-machine-runner-3-on-osx.adoc:97]: This is the macOS desktop UI, sometimes called an "Aqua" session in the apple docs.

**Comment by jrahme-cci** [comment]
@rosieyohannan I've made some updates to the Machine Runner 3.0 macos install doc, and added the doc for manual installation (without the package) for the macOS machine runner 3.0, as well as updated the langauge to remove the ambiguity around what you commented on. 

When you have a chance could you give the doc another review please?

**Comment by rosieyohannan** [comment]
@jrahme-cci I've made some style and formatting changes, and also updated the filename for the manual installation to use "installation" rather than "run". Let me know whether this is ok?

File: jekyll/_cci2/machine-runner-3-manual-installation.adoc [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+[#create-namespace-and-resource-class]
+== 1. Create namespace and resource class
+
+{% include snippets/runner/install-with-cli-steps.adoc %}
+
+[#download-circleci-runner]
+
+[#download-machine-runner]
+== 2. Download the CircleCI machine runner for macOS
```
Comment (rosieyohannan) [inline jekyll/_cci2/machine-runner-3-manual-installation.adoc:42]: @jrahme-cci I guess this "for macOS" is just an oversight, does it make sense to just remove that part so the section title is just `2. Download the CircleCI machine runner`?

Comment (jrahme-cci) [reply jekyll/_cci2/machine-runner-3-manual-installation.adoc]: It does. I've made the changes for that.

**Comment by jrahme-cci** [comment]
> @jrahme-cci I've made some style and formatting changes, and also updated the filename for the manual installation to use "installation" rather than "run". Let me know whether this is ok?

@rosieyohannan yep that looks good to me.

**Pull request merged**